src/configuration/configuration.ts

```diff
@@ -57,6 +57,21 @@ class ConfigurationClass {
         this[option] = vimOptionValue;
       }
     }
+
+    // Get the cursor type from vscode
+    const cursorStyleString = vscode.workspace.getConfiguration().get("editor.cursorStyle") as string;
+    this.userCursor = this.cursorStyleFromString(cursorStyleString);
+  }
+
+  private cursorStyleFromString(cursorStyle: string): vscode.TextEditorCursorStyle {
+    if (cursorStyle === 'line') {
+      return vscode.TextEditorCursorStyle.Line;
+    } else if (cursorStyle === 'block') {
+      return vscode.TextEditorCursorStyle.Block;
+    } else if (cursorStyle === 'underline') {
+      return vscode.TextEditorCursorStyle.Underline;
+    }
```

Comment:
Once we can upgrade to 1.10.0^ for the engine, I can add 2 more cases here for line-thin and underline-thin and it will work right away